site/examples/code-highlighting.tsx

```diff
@@ -234,20 +237,23 @@ const SetNodeToDecorations = () => {
     )
 
     editor.nodeToDecorations = nodeToDecorations
-  }, [editor.children])
+  }, [editor])
```

Comment:
We should be careful here, because `editor.children` might change even if `editor` doesn't.

Reply:
Good catch. That was significant here. What's confusing is that this function shouldn't have to know about editor.children. This function is setting up a custom `nodeToDecorations` map to optimize some of the other logic. I think it should be triggered by something more explicit than `editor.children` changing.

I can make an eslint exception for now, but the implementation should be thought through. The simplest change that doesn't require an exception would be to drop useMemo.